From fabe5dc3f3f545412b40d947af2b17e45ca17003 Mon Sep 17 00:00:00 2001
From: "Cao,Bing Bu" <bingbu.cao@intel.com>
Date: Tue, 5 May 2015 13:55:15 +0800
Subject: [PATCH 389/462] ov680: fix kw issues in ov680 driver

Change-Id: I09e99bd94377ee5dca6c19c8b8963720355bbfa8
Tracked-On: #JIMINAN-35074
Signed-off-by: Cao,Bing Bu <bingbu.cao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/54282
Build: IMIN (SC)
Reviewed-by: Gang A Hu <gang.a.hu@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov680.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
index 3ceb331..8743337 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
@@ -458,7 +458,10 @@ static int __power_ctrl(struct v4l2_subdev *sd, int on)
 	struct ov680_device *dev = to_ov680_device(sd);
 	int ret = 0;
 
-	if (!dev || !dev->platform_data)
+	if (!dev)
+		return -ENODEV;
+
+	if (!dev->platform_data)
 		return -ENODEV;
 
 	/* Non-gmin platforms use the legacy callback */
@@ -501,7 +504,10 @@ static int __gpio_ctrl(struct v4l2_subdev *sd, int on)
 #endif
 	struct ov680_device *dev = to_ov680_device(sd);
 	int ret = 0;
-	if (!dev || !dev->platform_data)
+	if (!dev)
+		return -ENODEV;
+
+	if (!dev->platform_data)
 		return -ENODEV;
 
 	/* Non-gmin platforms use the legacy callback */
-- 
1.9.1

